### PR TITLE
Enable linter `unparam` for test files

### PR DESCRIPTION
**Pull request by anishbista60**
## Change Overview

- Enable linter `unparam` for test files
## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [ ] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [ ] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test
- [ ] :building_construction: Build

## Issues 

- Ref #2962 

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- Ran `make golint`
- [ ] :green_heart: E2E

### CODE REVIEW

**Comment by anishbista60** [comment]
@pavannd1 would you mind to look into it ?

File: pkg/kopia/cli/repository/storage/s3/s3_test.go [hunk 1]
```diff
@@ -30,12 +30,12 @@ import (
...
 func TestNewS3(t *testing.T) { check.TestingT(t) }
 
-func newLocation(prefix, endpoint, region, bucket string, skipSSLVerify bool) internal.Location {
+func newLocation(prefix, endpoint, region string, skipSSLVerify bool) internal.Location {
 	return internal.Location{
 		"prefix":        []byte(prefix),
 		"endpoint":      []byte(endpoint),
 		"region":        []byte(region),
-		"bucket":        []byte(bucket),
+		"bucket":        []byte("bucket"),
```
Comment (viveksinghggits) [inline pkg/kopia/cli/repository/storage/s3/s3_test.go:38]: do we really need this change? what would happen if someone want to use another bucket?

**Review by viveksinghggits — CHANGES_REQUESTED** [review]
Looks good to me overall, left a question/concern.

File: pkg/kopia/cli/repository/storage/s3/s3_test.go (hunk 1 above)
Comment (anishbista60) [reply pkg/kopia/cli/repository/storage/s3/s3_test.go]: Make sense. The reason i hardcorded the value ist he bucket parameter of the `newLocation` function is always receiving the string "bucket" which is flagged as an issue by the `unparam` linter. Now, i fix it . 
Coud you please take look . 

Thank you

File: pkg/kopia/cli/repository/storage/s3/s3_test.go [hunk 2]
```diff
@@ -77,7 +80,7 @@ func toArgTests(s3tests []s3test) []inttest.ArgumentTest {
 var _ = check.Suite(&inttest.ArgumentSuite{Cmd: "cmd", Arguments: toArgTests([]s3test{
 	{
 		Name:     "NewS3",
-		Location: newLocation("prefix", "http://endpoint/path/", "region", "bucket", true),
+		Location: newLocation("prefix", "http://endpoint/path/", "region", "", true),
```
[the remark this answers is not in the document]
Comment (anishbista60) [reply pkg/kopia/cli/repository/storage/s3/s3_test.go:83]:
```
pkg/kopia/cli/repository/storage/s3/s3_test.go:33:44: newLocation - bucket always receives "bucket" (unparam)
func newLocation(prefix, endpoint, region, bucket string, skipSSLVerify bool) internal.Location {
```

[the remark this answers is not in the document]
Comment (viveksinghggits) [reply pkg/kopia/cli/repository/storage/s3/s3_test.go:83]: ok, let's ignore this linter in line just for this failure. that would imply that we don't need changes in `newLocation`.

**Comment by anishbista60** [comment]
@viveksinghggits Thankyou. please check it .

**Review by viveksinghggits — APPROVED** [review]

**Comment by anishbista60** [comment]
could you please run the test cases? and try to get it merge

**Pull request merged**